lib/svgo/css-style-declaration.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+var csstree  = require('css-tree'),
+    csstools = require('../css-tools');
+
+var CSSStyleDeclaration = function(node) {
+  this.parentNode = node;
+
+  this.properties = new Map();
+  this.length     = 0;
+
+  this.update(); // initialize with styles from style attribute
+};
+
+
+CSSStyleDeclaration.prototype.update = function() {
+  var self = this;
+  if(!this.parentNode.hasAttr('style')) {
+    return;
+  }
+  var inlineCssStr = this.parentNode.attr('style').value;
+  var cssoDeclarations = csstree.parse(inlineCssStr, {context: 'declarationList'});
+  cssoDeclarations.children.each(function(cssoDeclaration) {
+    var styleDeclaration = csstools.cssoToStyleDeclaration(cssoDeclaration);
+    self.setProperty(styleDeclaration.name, styleDeclaration.property.value, styleDeclaration.property.priority);
+  });
+};
+
+
+CSSStyleDeclaration.prototype.getCssText = function() {
+  var declarationList = csstree.parse('', {context: 'declarationList'});
```

Comment:
It's not clear why parse/translate needed here. Looks like this solution is good enough:
```js
CSSStyleDeclaration.prototype.getCssText = function() {
  return this.properties.map(function(property, propertyName) {
    var strImportant = property.priority === 'important' ? '!important' : '';
    return propertyName + ':' + property.value + strImportant;
  }).join(';');
};
```